### PR TITLE
Add C# hot reload support for generics to the "What's new in .NET 8" doc

### PR DESCRIPTION
**Pull request by lambdageek**
## Summary

Add a section to "What's new in .NET 8" that mentions support for hot reload of generic types and methods.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/whats-new/dotnet-8.md](https://github.com/dotnet/docs/blob/15ca7508a805bd14a96f42408b80382b9f9d043f/docs/core/whats-new/dotnet-8.md) | [What's new in .NET 8](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8?branch=pr-en-us-37761) |

### CODE REVIEW

**Review by mcumming — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by mikelle-rogers — APPROVED** [review]
Thanks @lambdageek!

**Comment by lambdageek** [comment]
It looks like I have permission to click the merge button myself.  Is that allowed?  Or do I need to wait for @gewarren to signoff?

File: docs/core/whats-new/dotnet-8.md
Comment (gewarren) [inline docs/core/whats-new/dotnet-8.md:1546]:
```suggestion
Starting in .NET 8, C# Hot Reload [supports modifying generic types and generic methods](https://devblogs.microsoft.com/dotnet/hot-reload-generics/). When you debug console, desktop, mobile, or WebAssembly applications with Visual Studio, you can apply changes to generic classes and generic methods in C# code or Razor pages. For more information, see the [full list of edits supported by Roslyn](https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md)
```

**Review by gewarren — APPROVED** [review]
I left a suggestion for consideration.

**Pull request merged**
